crates/re_viewer/src/ui/rerun_menu.rs

```diff
@@ -154,65 +156,11 @@ fn recordings_menu(ui: &mut egui::Ui, store_hub: &StoreHub, app_state: &mut AppS
             "<UNKNOWN>".to_owned()
         };
         if ui
-            .radio(
-                app_state.recording_id().as_ref() == Some(store_db.store_id()),
-                info,
-            )
-            .clicked()
-        {
-            app_state.set_recording_id(store_db.store_id().clone());
-        }
-    }
-}
-
-fn blueprints_menu(
-    ui: &mut egui::Ui,
-    app_id: &ApplicationId,
-    store_hub: &StoreHub,
-    app_state: &mut AppState,
-) {
-    let blueprint_dbs = store_hub
-        .blueprints()
-        .sorted_by_key(|store_db| store_db.store_info().map(|ri| ri.started))
-        .filter(|log| {
-            log.store_info()
-                .map_or(false, |ri| &ri.application_id == app_id)
-        })
-        .collect_vec();
-
-    if blueprint_dbs.is_empty() {
-        ui.weak("(empty)");
-        return;
-    }
-
-    ui.style_mut().wrap = Some(false);
-    for store_db in blueprint_dbs
-        .iter()
-        .filter(|log| log.store_kind() == StoreKind::Blueprint)
-    {
-        let info = if let Some(store_info) = store_db.store_info() {
-            if store_info.is_app_default_blueprint() {
-                format!("{} - Default Blueprint", store_info.application_id,)
-            } else {
-                format!(
-                    "{} - {}",
-                    store_info.application_id,
-                    store_info.started.format()
-                )
-            }
-        } else {
-            "<UNKNOWN>".to_owned()
-        };
-        if ui
-            .radio(
-                app_state.selected_blueprint_by_app.get(app_id) == Some(store_db.store_id()),
-                info,
-            )
+            .radio(active_recording == Some(store_db.store_id()), info)
             .clicked()
         {
-            app_state
-                .selected_blueprint_by_app
-                .insert(app_id.clone(), store_db.store_id().clone());
+            // TODO(jleibs): This is gross but necessary to avoid a deadlock
```

Comment:
I think what this todo is meaning to say is that having an extra field for delaying the setting is bad, but we'd instead want to queue it up as a command?
(because I was at first reading this as complaining about the copy)

Reply:
ah I see this is gone in the follow-up :)